src/python/pants/engine/internals/struct.py

```diff
@@ -315,3 +316,14 @@ def dependencies(self):
 
         :rtype: tuple
         """
+
+
+class TargetAdaptor(StructWithDeps):
+    """A Struct to hold a Target's values before being converted into the Target API."""
+
+
+@dataclass(frozen=True)
+class HydratedTargetAdaptor:
```

Comment:
Is this intermediate class necessary anymore, or could we request `TargetAdaptor` directly now that that class is concrete?

Reply:
Maybe not. Although I like it to distinguish between a raw `TargetAdaptor` vs. one that's been hydrated, e.g. the `dependencies` field being normalized.

Reply:
Well, it's a bit of misnomer now. "hydrated" used to mean "dependencies have been inlined/expanded" (the recursive stuff we used to do there). Now it's pure parsing, which isn't particularly heavy. Anywho.